v2.0/show-grants.md

```diff
@@ -21,135 +21,157 @@ No [privileges](privileges.html) are required to view privileges granted to user
+> SHOW GRANTS ON TABLE test_roles.employees FOR system_ops;
 ~~~
 ~~~ shell
-+-------+------------+------------+
-| Table |    User    | Privileges |
-+-------+------------+------------+
-| t1    | betsyroach | DELETE     |
-| t1    | maxroach   | DELETE     |
-| t2    | betsyroach | DELETE     |
-| t2    | maxroach   | DELETE     |
-+-------+------------+------------+
-(4 rows)
++------------+--------+-----------+------------+------------+
+|  Database  | Schema |   Table   |    User    | Privileges |
++------------+--------+-----------+------------+------------+
+| test_roles | public | employees | system_ops | CREATE     |
++------------+--------+-----------+------------+------------+
 ~~~
 
 **All tables, all users:**
 
+{% include copy-clipboard.html %}
 ~~~ sql
-> SHOW GRANTS ON TABLE db1.*;
+> SHOW GRANTS ON TABLE test_roles.*;
 ~~~
 
 ~~~ shell
-+-------+------------+------------+
-| Table |    User    | Privileges |
-+-------+------------+------------+
-| t1    | betsyroach | DELETE     |
-| t1    | henryroach | DELETE     |
-| t1    | maxroach   | DELETE     |
-| t1    | root       | ALL        |
-| t1    | sallyroach | DELETE     |
-| t2    | betsyroach | DELETE     |
-| t2    | henryroach | DELETE     |
-| t2    | maxroach   | DELETE     |
-| t2    | root       | ALL        |
-| t2    | sallyroach | DELETE     |
-| t3    | root       | ALL        |
-| t4    | maxroach   | CREATE     |
-| t4    | root       | ALL        |
-| t5    | maxroach   | CREATE     |
-| t5    | root       | ALL        |
-+-------+------------+------------+
-(15 rows)
-~~~
-
-**All tables, specific users:**
++------------+--------+-----------+------------+------------+
+|  Database  | Schema |   Table   |    User    | Privileges |
++------------+--------+-----------+------------+------------+
+| test_roles | public | employees | admin      | ALL        |
+| test_roles | public | employees | root       | ALL        |
+| test_roles | public | employees | system_ops | CREATE     |
++------------+--------+-----------+------------+------------+
+~~~
 
+**All tables, specific users or roles:**
+
+{% include copy-clipboard.html %}
 ~~~ sql
-> SHOW GRANTS ON TABLE db1.* FOR maxroach, betsyroach;
+> SHOW GRANTS ON TABLE test_roles.* FOR system_ops;
 ~~~
 
 ~~~ shell
-+-------+------------+------------+
-| Table |    User    | Privileges |
-+-------+------------+------------+
-| t1    | betsyroach | DELETE     |
-| t1    | maxroach   | DELETE     |
-| t2    | betsyroach | DELETE     |
-| t2    | maxroach   | DELETE     |
-| t4    | maxroach   | CREATE     |
-| t5    | maxroach   | CREATE     |
-+-------+------------+------------+
-(6 rows)
++------------+--------+-----------+------------+------------+
+|  Database  | Schema |   Table   |    User    | Privileges |
++------------+--------+-----------+------------+------------+
+| test_roles | public | employees | system_ops | CREATE     |
++------------+--------+-----------+------------+------------+
 ~~~
```

Comment:
We should document `SHOW GRANTS ON ROLE` and its variations:
- that statement exactly: show all role memberships
- `.. ON ROLE <role1, role2, ....>`: show memberships in specific roles
- `.. ON ROLE FOR <name1, name2, ...>`: show which roles `name1` and `name2` are members of
- `.. ON ROLE <role1, role2, ..> FOR `<name1, name2, ...>`: show specific roles and members.